### PR TITLE
Fetch recent activities from graphql

### PR DESCRIPTION
**Pull request by subinasr**
- Addresses #2512 
- Depends on https://github.com/the-deep/server/pull/1128/files

## Changes

* Use graphql to fetch recent activities

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

**Review by tnagorra — APPROVED** [review]

File: app/views/Home/RecentActivity/ActivityItem/index.tsx [hunk 1]
```diff
@@ -1,70 +1,137 @@
...
+        if (activity?.type === 'LEAD') {
+            return (generateString(
+                'a {link} on',
+                {
+                    link: (
+                        <Link
+                            className={styles.link}
+                            to={generatePath(routes.entryEdit.path, {
+                                projectId: activity?.project?.id,
+                                leadId: '1',
```
Comment (AdityaKhatri) [inline app/views/Home/RecentActivity/ActivityItem/index.tsx]: Lets use the actual leadId

File: app/views/Home/RecentActivity/ActivityItem/index.tsx [hunk 2]
```diff
@@ -1,70 +1,137 @@
...
-        type,
+        activity,
     } = props;
 
+    const typeDisplayText = useMemo(() => {
+        const editEntryLink = {
+            pathname: (generatePath(routes.entryEdit.path, {
+                projectId: activity?.project?.id,
+                // TODO: Remove this
+                leadId: activity?.leadId ?? '1',
```
Comment (AdityaKhatri) [inline app/views/Home/RecentActivity/ActivityItem/index.tsx]: If there is no leadId, lets not make it a link

File: app/views/Home/RecentActivity/ActivityItem/index.tsx [hunk 3]
```diff
@@ -1,70 +1,137 @@
...
+            })),
+            state: {
+                // TODO: Remove this
+                entryServerId: ((activity?.type === 'ENTRY')
+                     || (activity?.type === 'ENTRY_COMMENT')) && activity?.leadId,
```
Comment (AdityaKhatri) [inline app/views/Home/RecentActivity/ActivityItem/index.tsx]: I think this should be entyId

File: app/views/Home/RecentActivity/ActivityItem/index.tsx (hunk 1 above)
Comment (AdityaKhatri) [reply app/views/Home/RecentActivity/ActivityItem/index.tsx]: Also, send the user to #primary-tagging

File: app/views/Home/RecentActivity/ActivityItem/index.tsx [hunk 4]
```diff
@@ -1,70 +1,137 @@
...
+            return (generateString(
+                'an {link} on',
+                {
+                    link: (
+                        <Link
+                            className={styles.link}
+                            to={generatePath(routes.entryEdit.path, {
+                                projectId: activity?.project?.id,
+                                leadId: '1',
+                                entryId: '1',
```
Comment (AdityaKhatri) [inline app/views/Home/RecentActivity/ActivityItem/index.tsx]: Lets use real entry and leadIds

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Fix comments

**Pull request merged**